CMakeLists.txt

```diff
@@ -100,201 +54,82 @@ else ()
-                       ABSOLUTE CACHE)
+# Look for the opm-data repository; if found the variable
+# HAVE_OPM_DATA will be set to true.
+include(Findopm-data)
 
-if (EXISTS ${OPM_DATA_ROOT}/norne/NORNE_ATW2013.DATA)
-    set(HAVE_OPM_DATA TRUE)
-else()
-    set(HAVE_OPM_DATA FALSE)
-    message(WARNING "opm-data not found - skipping some integration tests.")
-endif ()
-
-# Setup static analysis
-include( cmake/Modules/UseStaticAnalysis.cmake )
-list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
-setup_static_analysis_tools()
+# not the same location as most of the other projects; this hook overrides
+macro (dir_hook)
+  set (doxy_dir "docs")
+endmacro (dir_hook)
 
-add_subdirectory(lib/json)
-add_subdirectory(lib/eclipse)
-add_subdirectory(applications)
+# list of prerequisites for this particular project; this is in a
+# separate file (in cmake/Modules sub-directory) because it is shared
+# with the find module
+include (${project}-prereqs)
 
-set(OPM_PARSER_PREFIX "${CMAKE_INSTALL_PREFIX}")
-set(OPM_PARSER_MODULE_PATH "${OPM_PARSER_PREFIX}/share/dune/cmake/modules")
-set(OPM_PARSER_INCLUDEDIRS "${OPM_PARSER_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}")
-set(OPM_PARSER_LIBDIRS "${OPM_PARSER_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
-set(OPM_PARSER_INSTALLED ON)
-get_target_property(ecl_lib ecl LOCATION)
-get_target_property(libtype opmparser TYPE)
-if("${libtype}" STREQUAL "STATIC_LIBRARY")
-  set(suffix ${CMAKE_STATIC_LIBRARY_SUFFIX})
-  set(prefix ${CMAKE_STATIC_LIBRARY_PREFIX})
-else()
-  set(suffix ${CMAKE_SHARED_LIBRARY_SUFFIX})
-  set(prefix ${CMAKE_SHARED_LIBRARY_PREFIX})
-endif()
-set(OPM_PARSER_LIBRARIES opmparser opmjson ${ecl_lib} ${Boost_FILESYSTEM_LIBRARY} ${Boost_REGEX_LIBRARY})
-set(OPM_PARSER_LIBRARY ${CMAKE_INSTALL_PREFIX}/lib/${prefix}opmparser${suffix})
-set(ECL_DIR "${OPM_PARSER_PREFIX}/share/cmake/ecl")
-configure_file(opm-parser-config.cmake.in
-  "${PROJECT_BINARY_DIR}/install/opm-parser-config.cmake" @ONLY)
+# read the list of components from this file (in the project directory);
+# it should set various lists with the names of the files to include
+include (CMakeLists_files.cmake)
 
-set(OPM_PARSER_PREFIX "${PROJECT_SOURCE_DIR}")
-set(OPM_PARSER_INCLUDEDIRS "${OPM_PARSER_PREFIX}/lib/eclipse/include;${OPM_PARSER_PREFIX}/lib/json/include;${PROJECT_BINARY_DIR}/lib/eclipse/include")
-set(OPM_PARSER_MODULE_PATH "${OPM_PARSER_PREFIX}/cmake/modules")
-set(OPM_PARSER_LIBDIRS "${CMAKE_BINARY_DIR}/lib/eclipse;${CMAKE_BINARY_DIR}/lib/json")
-set(OPM_PARSER_LIBRARY ${CMAKE_BINARY_DIR}/lib/eclipse/${prefix}opmparser${suffix})
-set(OPM_PARSER_INSTALLED OFF)
-set(ECL_DIR "${ecl_DIR}")
-configure_file(opm-parser-config.cmake.in
-  "${PROJECT_BINARY_DIR}/opm-parser-config.cmake" @ONLY)
-
-configure_file(opm-parser-config-version.cmake.in
-  "${PROJECT_BINARY_DIR}/opm-parser-config-version.cmake" @ONLY)
+macro (config_hook)
+  list(APPEND opm-parser_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}/lib/eclipse/include
```

Comment:
Does this have  side effect on opm-parser-config.cmake? In that case does it work as expected for installed modules?

Reply:
yeah, didn't consider that. it works with installs but adds garbage include directories. 

it's a bit tricky to get right until we use targets, since not having them in there will kill the sibling/dc builds. so i had to do the EXTRA_CODE dance for the time being.